handlers/handlers.go

```diff
@@ -40,68 +42,87 @@ func setStatusCode(req *http.Request, w http.ResponseWriter, err error) {
 	w.WriteHeader(status)
 }
 
-func forwardFlorenceTokenIfRequired(req *http.Request) *http.Request {
-	if len(req.Header.Get(common.FlorenceHeaderKey)) > 0 {
-		ctx := common.SetFlorenceIdentity(req.Context(), req.Header.Get(common.FlorenceHeaderKey))
-		return req.WithContext(ctx)
-	}
-	return req
-}
-
-//GeographyRender ...
-func GeographyRender(rend RenderClient) http.HandlerFunc {
+//GeographyHomepageRender ...
+func GeographyHomepageRender(rend RenderClient) http.HandlerFunc {
 
 	return func(w http.ResponseWriter, req *http.Request) {
 		var page geographyHomepage.Page
+		ctx := req.Context()
 
-		resp, err := http.Get(`https://api.dev.cmd.onsdigital.co.uk/v1/code-lists` + localAuthority)
+		homePageLink := url + `/code-lists?type=geography`
+
+		resp, err := http.Get(homePageLink)
 		if err != nil {
+			err = errors.Wrap(err, "error rendering homepage - failed to get data from the code-lists api")
+			log.ErrorCtx(ctx, err, log.Data{"error": err, "url": homePageLink})
 			setStatusCode(req, w, err)
 			return
 		}
+		defer resp.Body.Close()
+
 		b, err := ioutil.ReadAll(resp.Body)
 		if err != nil {
+			err = errors.Wrap(err, "error reading the .Body data")
+			log.ErrorCtx(ctx, err, log.Data{"error": err, "url": homePageLink})
 			setStatusCode(req, w, err)
 			return
 		}
 		var codelistresults models.CodeListResults
 		err = json.Unmarshal(b, &codelistresults)
 		if err != nil {
-			setStatusCode(req, w, err)
-			return
-		}
-		var codelist models.CodeList
-		err = json.Unmarshal(b, &codelist)
-		if err != nil {
+			err = errors.Wrap(err, "error unmarshaling from .CodeListResults")
+			log.ErrorCtx(ctx, err, log.Data{"error": err, "url": homePageLink})
 			setStatusCode(req, w, err)
 			return
 		}
 
-		geographyTypes := ""
+		var geographyTypes []geographyHomepage.Items
 		for i := range codelistresults.Items {
-			log.Debug("test", log.Data{
-				"geographyTypesTest": codelistresults.Items[i],
-			})
-
-			geographyTypes = geographyTypes + codelistresults.Items[i].Name
-
-		}
 
-		page.Data.AreaTypes = []geographyHomepage.AreaType{
-			{Name: "Countries"},
-			{Name: "Regions"},
-			{Name: geographyTypes},
+			geographyTypesID := codelistresults.Items[i].Links.Self.ID
+			typesListLink := codelistresults.Items[i].Links.Editions.Href
+
+			resp, err := http.Get(typesListLink)
+			if err != nil {
+				err = errors.Wrap(err, "error rendering geography types list - failed to get data from the code-lists api")
+				log.ErrorCtx(ctx, err, log.Data{"error": err, "url": typesListLink})
+				setStatusCode(req, w, err)
+				return
+			}
+			defer resp.Body.Close()
+			b, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				err = errors.Wrap(err, "error reading the .Body data")
+				log.ErrorCtx(ctx, err, log.Data{"error": err, "url": typesListLink})
+				setStatusCode(req, w, err)
+				return
+			}
+			var codelistresults models.CodeListResults
```

Comment:
This should probably be a different model which will map the actual editions endpoint response, rather than reusing the same struct across multiple endpoints and only using some of the fields in each case.

Even though the results struct is of a similar structure and the concept of links is similar across both endpoints; since the editions response has more than just a label and no ID, and the code list response will never have a label, tying them together like this could cause problems in the future. 

What if both endpoints had a new field called "filterable" but on one end point it was required and the other it wasn't? In the short term, if we did want to use the same struct for both they should be more clearly named to make it clear that this is used for both cases - but then you get clunky names.

I would consider splitting the models for each response. There is almost certainly a way to reduce the duplication by defining a common struct for the API list response fields like limit, offset, etc and then embedding that in the specific struct which provides the typed list e.g.:

```
type EditionsList struct {
Items []Editions `json:items`
apiResponse
}

type apiResponse struct {
Count      int        `json:"count"`
	Offset     int        `json:"offset"`
	Limit      int        `json:"limit"`
	TotalCount int        `json:"total_count"`
}
```